app/src/main/java/org/oppia/android/app/onboarding/OnboardingFragmentPresenter.kt

```diff
@@ -36,13 +84,197 @@ class OnboardingFragmentPresenter @Inject constructor(
+        oppiaLogger.e(
+          "OnboardingFragment",
+          "Failed to retrieve system language locale.",
+          result.error
+        )
+        OppiaLanguage.ENGLISH
+      }
+      is AsyncResult.Pending -> OppiaLanguage.ENGLISH
+    }
+  }
+
+  private fun getSupportedLanguages() {
+    translationController.getSupportedAppLanguages().toLiveData().observe(
+      fragment,
+      { result ->
+        when (result) {
+          is AsyncResult.Success -> {
+            val supportedLanguages = mutableListOf<String>()
+            result.value.map {
+              supportedLanguages.add(appLanguageResourceHandler.computeLocalizedDisplayName(it))
+              onboardingAppLanguageViewModel.setSupportedAppLanguages(supportedLanguages)
+            }
+          }
+          is AsyncResult.Failure -> {
+            oppiaLogger.e(
+              "OnboardingFragment",
+              "Failed to retrieve supported language list.",
+              result.error
+            )
+          }
+          is AsyncResult.Pending -> {}
+        }
+      }
+    )
+  }
+
+  private fun subscribeToGetProfileList() {
+    existingProfiles.observe(
+      fragment,
+      { profilesList ->
+        if (!profilesList.isNullOrEmpty()) {
+          retrieveProfileId(profilesList)
+        } else {
+          createDefaultProfile()
+        }
+      }
+    )
+  }
+
+  private val existingProfiles: LiveData<List<Profile>> by lazy {
+    Transformations.map(
+      profileManagementController.getProfiles().toLiveData(),
+      ::processGetProfilesResult
+    )
+  }
+
+  private fun processGetProfilesResult(profilesResult: AsyncResult<List<Profile>>): List<Profile> {
+    val profileList = when (profilesResult) {
+      is AsyncResult.Failure -> {
+        oppiaLogger.e(
+          " OnboardingFragment", "Failed to retrieve the list of profiles", profilesResult.error
+        )
+        emptyList()
+      }
+      is AsyncResult.Pending -> emptyList()
+      is AsyncResult.Success -> profilesResult.value
+    }
+
+    return profileList
+  }
+
+  private fun createDefaultProfile() {
+    profileManagementController.addProfile(
+      name = "Admin", // TODO(#4938): Refactor to empty name once proper admin profile creation flow
+      // is implemented.
+      pin = "",
+      avatarImagePath = null,
+      allowDownloadAccess = allowDownloadAccess,
+      colorRgb = -10710042,
+      isAdmin = true
+    ).toLiveData()
+      .observe(
+        fragment,
+        { result ->
+          when (result) {
+            is AsyncResult.Success -> subscribeToGetProfileList()
+            is AsyncResult.Failure -> oppiaLogger.e(
+              "OnboardingFragment", "Error creating the default profile", result.error
+            )
+            is AsyncResult.Pending -> {}
+          }
+        }
+      )
+  }
+
+  private fun retrieveProfileId(profileList: List<Profile>) {
+    profileId = profileList.firstOrNull()?.id ?: ProfileId.getDefaultInstance()
```

Comment:
Suggest ininlining this method above, and reworkiong the logic to avoid the double "if empty" checks happening here (for robustness and simplicity).